### PR TITLE
package installation cleanup

### LINKED ISSUE #7163: bug: Permission problems
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

We are using Localstack with our Openshift Cluster. The default behavior is, that every pod gets a random UID.
With the current Localstack Image we get the following error:

`/usr/local/bin/docker-entrypoint.sh: line 73: /var/lib/localstack/logs/localstack_infra.log: Permission denied`

Reproducible with following command: 

`docker run -ti --rm --user 6666 localstack/localstack:1.2.0`

From inspecting the dockerfile I would expect "rwx" permissions for user/group/others but in the running container I can see that group and others have only "rx" permissions. I think the permissions will be changed because the original directory /var/lib/localstack will be copied into a volume. 

Is it possible to remove this volume from dockerfile?

### Expected Behavior

`docker run -ti --rm --user 6666 localstack/localstack:1.2.0`

starts localstack without permission problems

### How are you starting LocalStack?

Custom (please describe below)

### Steps To Reproduce

localstack will be startet es container from put Gitllab Runner (Kubernetes/Openshift) because we use it as service in a pipeline job.

### Environment

```markdown
- OS:RedHat
- LocalStack:1.2.0 (latest)
```

### PR DESCRIPTION
**Pull request by alexrashed**
This PR performs a final cleanup of the legacy code in community, which was left after the migration to the new package abstraction introduced in https://github.com/localstack/localstack/pull/6783.
It contains the following changes:
- Remove the `init` make target and its usage in the `Dockerfile`.
  - Instead, the `Dockerfile` now uses `lpm` and explicitly states the packages which should be contained in the image by default.
  - The installation of `elasticmq` (which is used in the legacy sqs provider) is not migrated, since it has been deprecated quite some time ago (and elasticmq will be installed on demand anyways).
- Removes the `install` hook (since it became obsolete, as it was only called by `make init`).
- Migrates the remaining - and still used - constants in `localstack.services.install` to their proper locations.
- Migrate the remaining utility functions in `localstack.services.install` to `localstack.utils.archives`.
- Finally removes the `localstack.services.install` module.
- Removes the `Vagrantfile`.
  - This is not directly related to this PR, but the `Vagrantfile` is not functional anymore and is not reported to be used by any collaborators anymore.
- Sets 777 on `/var/lib/localstack` to allow running localstack with a non-root user. Fixes #7163 /cc @dfangl 

TODO:
- [ ] Either remove the last commit (creating a temporary install module again, redirecting to the new locations, to fix the tests), or remove it in an upcoming PR.

### CODE REVIEW

**Review by thrau — APPROVED** [review]
LGTM!

File: Dockerfile [hunk 1]
```diff
@@ -251,6 +246,26 @@ RUN (PIP_ARGS=$([[ "$LOCALSTACK_PRE_RELEASE" == "1" ]] && echo "--pre" || true);
...
+      local-kms \
+      mqtt \
+      postgres \
+      redis \
+      stepfunctions \
+      stepfunctions-local \
+      timescaledb && \
+    rm -rf /tmp/localstack/* && \
+    rm -rf /var/lib/localstack/cache/* && \
+    chmod -R 777 /var/lib/localstack
```
Comment (dfangl) [inline Dockerfile:266]: Not too sure if this approach is not too aggressive in general, but it's ok for me :)

File: Dockerfile [hunk 2]
```diff
@@ -259,7 +274,7 @@ ENV LOCALSTACK_BUILD_DATE=${LOCALSTACK_BUILD_DATE}
 ENV LOCALSTACK_BUILD_GIT_HASH=${LOCALSTACK_BUILD_GIT_HASH}
 ENV LOCALSTACK_BUILD_VERSION=${LOCALSTACK_BUILD_VERSION}
 
-# clean up some libs (e.g., Maven should be no longer required after "make init" has completed)
+# clean up some libs (e.g., Maven should be no longer required after initial installation has completed)
 RUN rm -rf /usr/share/maven
```
Comment (dfangl) [inline Dockerfile:278]: Just noticed that. Shouldn't we include this in the lpm install call? Removing it in a separate layer seems wasteful storage wise.

**Review by dfangl — APPROVED** [review]
Lambda / dockerfile permission changes LGTM!

**Pull request merged**